File: bc.cpp
#include "bc.h"

void bc(Matrix<double>& u)
{
	struct test_cases* pt_test = get_tests();

	int m = u.rows();
	int n = u.cols() - 2*pt_test->gc;

	for (int j = 1; j <= pt_test->gc; j++)
	{
		for (int i = 1; i <= m; i++)
		{
			u(i, j) = u(i, 2*pt_test->gc + 1 - j);
			u(i, pt_test->gc + n + j) = u(i, pt_test->gc + 1 + n - j);			
		}		
	}
}

File: bc.m
function u=bc(u)

%int:n,m,i,j,gc
%matrix:u

global gc;

m=size(u, 1);
n=size(u, 2)-2*gc;

for j=1:gc
  for i=1:m
%     u(i, j)=1e10;
%     u(i, gc+n+j)=1e10;
    u(i, j)=u(i, 2*gc+1-j);
    u(i, gc+n+j)=u(i, gc+1+n-j);
  end
end